src/Microsoft.OData.Core/UriParser/SemanticAst/ConstantNode.cs

```diff
@@ -50,7 +50,9 @@ public ConstantNode(object constantValue, string literalText)
         /// <exception cref="System.ArgumentNullException">Throws if the input literalText is null.</exception>
         public ConstantNode(object constantValue, string literalText, IEdmTypeReference typeReference)
         {
-            ExceptionUtils.CheckArgumentStringNotNullOrEmpty(literalText, "literalText");
+            // Commented the validation line below since we should be able to create a 
+            // ConstantNode whose LiteralText is an empty string '' or String.Empty
+            //ExceptionUtils.CheckArgumentStringNotNullOrEmpty(literalText, "literalText");

```

Comment:
@KenitoInc Should we not retain the check for null? In addition the method documentation string has `/// <exception cref...`

Reply:
Plus if we decide to abandon the null/empty string check altogether, I don't think it's necessary to retain the commented lines

Reply:
My concern with doing a null check is because `LiteralText` is still allowed to be `null`. Look at the `<summary>` below.
https://github.com/OData/odata.net/blob/871d97aec1fab62dc590d4f4f10ba2bab8901670/src/Microsoft.OData.Core/UriParser/SemanticAst/ConstantNode.cs#L81-L84
Any thoughts @mikepizzo @habbes ?

Reply:
I'm okay removing the null check given that we have an overload that omits the literalText and already handle the resulting null case.

As Christof points out, the current behavior to create an empty string constant node is to pass in the literalText "/"/"".  Is the correct fix for an empty string in the In operator to modify the ConstantNode to allow an empty string, or to pass in "/"/"" as the literal text for an empty string from the InBinder?

Reply:
I agree with Christof, The correct way is to pass an empty string. I have updated the fix and retained the Null check